crates/biome_js_analyze/src/analyzers/nursery/no_unused_private_class_members.rs

```diff
@@ -0,0 +1,299 @@
+        .filter_map(TsPropertyParameter::cast)
+        .filter(|param| is_private(param.syntax()))
+        .filter_map(|param| {
+            let param_name = param
+                .formal_parameter()
+                .ok()?
+                .as_js_formal_parameter()?
+                .binding()
+                .ok()?
+                .as_any_js_binding()?
+                .as_js_identifier_binding()?
+                .name_token()
+                .ok()?;
+
+            let text_token = param_name.token_text_trimmed();
+            let text_range = param_name.text_trimmed_range();
+
+            let state = RuleState {
+                range: text_range,
+                member: AnyPrivateMember::TsPropertyParameter(param),
+            };
+
+            Some((text_token, state))
+        });
+
+    class_declaration
+        .members()
+        .iter()
+        .filter(|member| is_private(member.syntax()))
+        .filter_map(|member| {
+            let text_token = get_member_text_token(&member)?;
+            let text_range = get_member_range(&member)?;
+            let state = RuleState {
+                range: text_range,
+                member: AnyPrivateMember::AnyJsClassMember(member),
+            };
+
+            Some((text_token, state))
+        })
+        .chain(constructor_params)
+        .collect()
+}
+
+fn is_private(syntax: &SyntaxNode<JsLanguage>) -> bool {
+    syntax.descendants().any(|node| {
+        let is_es_private = JsPrivateClassMemberName::cast_ref(&node).is_some();
+        let is_ts_private = TsAccessibilityModifier::cast_ref(&node)
+            .is_some_and(|accessibility| accessibility.is_private());
+
+        is_es_private || is_ts_private
+    })
+}
+
+/// Check whether the js name is in a write only assignment.
+fn is_write_only(js_name: &AnyJsName) -> bool {
+    let grand_parent = js_name.syntax().grand_parent();
+    let is_assignment_expression = grand_parent
+        .kind()
+        .is_some_and(|kind| matches!(kind, JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION));
+
+    // Skip every non assignment expression
+    if !is_assignment_expression {
+        return false;
+    }
+
+    if let Some(grand_parent) = grand_parent {
+        if let Some(parent) = js_name.syntax().parent() {
+            if let Some(assignment_expression) = JsAssignmentExpression::cast(grand_parent) {
+                let left = assignment_expression.left().ok();
+
+                if left.is_some_and(|left| !is_node_equal(left.syntax(), &parent)) {
+                    return false;
+                }
+
+                if !matches!(
+                    assignment_expression.operator(),
+                    Ok(JsAssignmentOperator::Assign)
+                ) {
+                    let kind = assignment_expression.syntax().parent().kind();
+                    return kind
+                        .is_some_and(|kind| matches!(kind, JsSyntaxKind::JS_EXPRESSION_STATEMENT));
+                }
+            }
+        }
+    }
```

Comment:
We could merge these branches?

We check twice if grand parent is an assignment... This looks non-optimal?

Also, since we use both, pare,t and grand-parent, we could extract the grand-parent from the parent.